### DISCUSSION TITLE
After creating the app, modules are not found and React must be imported in all modules.

### DISCUSSION THREAD

**Discussion by alexisInowuDev** [callstack/react-native-builder-bob#559] [Q&A]
After successfully creating the app with the command `npx create-react-native-library@latest awesome-library`, The app cannot run because modules are not found:

![Screenshot 2024-05-27 at 11 57 48 a m](https://github.com/callstack/react-native-builder-bob/assets/164578262/307b531f-5316-405c-9c35-85c0e9ea187b)

And React must be imported everywhere?
![Screenshot 2024-05-27 at 11 58 08 a m](https://github.com/callstack/react-native-builder-bob/assets/164578262/eb74cb6a-a818-4e76-8a62-befd790e47c5)

Am I doing something wrong?

**Comment by satya164** [comment]
Do you get any errors when running `yarn typecheck` in the terminal? If not then it's a problem with the editor setup. Also, try disabling the error lens extension and see if errors still appear.

**Reply by alexisInowuDev** [reply]
![Screenshot 2024-05-29 at 11 15 53 a m](https://github.com/callstack/react-native-builder-bob/assets/164578262/091b7707-0dcd-4753-8b71-e4389f97f4bf)
I get those errors, i tried to follow the development flow first, by running `yarn` at the root, then `yarn example start`, at root again. That failed, then i got the error above with `yarn typecheck`. 

Btw, thanks for replying so fast!

**Reply by satya164** [reply]
@alexisInowuDev there was a change in expo that broke template generation. please try again with the latest version and those files shouldn't exist.

**Reply by alexisInowuDev** [reply]
Working well now! Thank you very much!
But i have a couple of questions, in the example app, why does it not create an ios and android folder?
And also, I see that this tool uses expo, the library i create, will users who do not use Expo be able to use my published library?

Sorry if the questions seem quite obvious, im new to React Native development.

**Reply by satya164** [reply]
> in the example app, why does it not create an ios and android folder

If you chose the JS library option then we assume you don't need to modify native code. If you need access to native code for some reason, you can run `expo prebuild`. Though, it shouldn't be necessary in most cases.

> I see that this tool uses expo, the library i create, will users who do not use Expo be able to use my published library

Yes. Expo is only to make your development experience easier. As long as you don't import anything from Expo in your library code, anyone can use it.
